Sources/Basic/OutputByteStream.swift

```diff
@@ -55,32 +58,108 @@ public class OutputByteStream: OutputStream {
         return buffer.count
     }
 
-    /// The contents of the output stream.
-    ///
-    /// This method implicitly flushes the stream.
-    public var bytes: ByteString {
-        flush()
-        return ByteString(self.buffer)
+    /// Currently available buffer size.
+    private var availableBufferSize: Int {
+        return max(buffer.capacity, OutputByteStream.bufferSize) - buffer.count
```

Comment:
The max is wrong here, capacity will never be more than buffersize unless there is a bug in code.
I could not find an initialiser on Array which takes capacity. We could probably do something like 
```swift
self.buffer = [UInt8](repeating: 0, count: buffersize)
self.buffer.removeAll(keepingCapacity: true)
```
Not sure if that is expensive to do? other option is using `UnsafeMutablePointer<UInt8>` and manually manage memory